### PR TITLE
Add code and examples

### PR DESCRIPTION
**Pull request by cgrushko**

### CODE REVIEW

**Comment by cgrushko** [comment]
@mhlopko

File: cc_configure_make/def.bzl [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    out_lib = ctx.actions.declare_file("{}.a".format(ctx.attr.name))
+    outputs = [out_includes, out_lib]
+
+    cpp_fragment = ctx.fragments.cpp
+    compiler_options = [] # cpp_fragment.compiler_options(ctx.features)
+    c_options = compiler_options + cpp_fragment.c_options
+    cxx_options = compiler_options + cpp_fragment.cxx_options(ctx.features)
+
+    CFLAGS = "\"{}\"".format(' '.join(c_options))
+    CXXFLAGS = "\"{}\"".format(' '.join(cxx_options))
```
Comment (mhlopko) [inline cc_configure_make/def.bzl]: FYI, but there's a *lot* of env vars that we should support (https://www.gnu.org/software/make/manual/html_node/Implicit-Variables.html). We will add them as we go.

File: cc_configure_make/def.bzl [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+def _cc_configure_make_impl(ctx):
+    out_includes = ctx.actions.declare_directory(ctx.attr.name + "-includes.h")
+    out_lib = ctx.actions.declare_file("{}.a".format(ctx.attr.name))
+    outputs = [out_includes, out_lib]
+
+    cpp_fragment = ctx.fragments.cpp
```
Comment (mhlopko) [inline cc_configure_make/def.bzl]: Skylark api on the CppConfiguration is being removed as we speak, see cl/194106058 for an example migration.

File: cc_configure_make/def.bzl (hunk 1 above)
Comment (mhlopko) [reply cc_configure_make/def.bzl]: Hmm but we should specify `CC` and `CXX` at least, and once https://docs.google.com/document/u/1/d/1g91BWJITcYw_X-VxsDC0VgUn5E9g0kRBGoBSpoO41gA/edit is done (O(weeks)) we can specify different tools for those

File: cc_configure_make/def.bzl [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    compiler_options = [] # cpp_fragment.compiler_options(ctx.features)
```
[the remark this answers is not in the document]
Comment (cgrushko) [reply cc_configure_make/def.bzl]: This trips `c-ares`, so I want to add it only after I have an idea how to handle it.

File: cc_configure_make/def.bzl (hunk 1 above)
Comment (cgrushko) [reply cc_configure_make/def.bzl]: How will the code look like here?

File: cc_configure_make/def.bzl (hunk 2 above)
Comment (cgrushko) [reply cc_configure_make/def.bzl]: Is there an external version?

File: cc_configure_make/def.bzl (hunk 1 above)
Comment (mhlopko) [reply cc_configure_make/def.bzl]: roughly like this: cc_common.configure_features(cc_toolchain).get_command_line(cc_common.c_compile_action_name, cc_common.get_compile_variables(....))

File: cc_configure_make/def.bzl (hunk 2 above)
Comment (mhlopko) [reply cc_configure_make/def.bzl]: I'm not aware of any yet.

**Pull request merged**